ZoomEye - If added togheter with other engines, sometimes fail with the same error

When performing multiple searches using the -e zoomeye command along with other search engines, an error occasionally occurs with the following panic message:

```
panic: interface conversion: interface {} is string, not float64
goroutine 58 [running]:
github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).query(0x0?, {0x1286f8b?, 0x0?}, 0x0?, 0x0?, 0x0?)
	/home/runner/work/uncover/uncover/uncover/agent/zoomeye/zoomeye.go:94 +0x4e5
github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).Query.func1()
	/home/runner/work/uncover/uncover/uncover/agent/zoomeye/zoomeye.go:42 +0x10b
created by github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).Query
	/home/runner/work/uncover/uncover/uncover/agent/zoomeye/zoomeye.go:31 +0xcc
```

### uncover version:
1.0.3

### Steps To Reproduce:
-r -silent -e shodan,censys,zoomeye -q "apache"

Note that this issue occurs intermittently, not consistently with every search.

--- a/sources/agent/zoomeye/zoomeye.go
+++ b/sources/agent/zoomeye/zoomeye.go
@@ -91,7 +91,7 @@ func (agent *Agent) query(URL string, session *sources.Session, zoomeyeRequest *
 		}
 		if portinfo, ok := zoomeyeResult["portinfo"]; ok {
 			if port, ok := portinfo.(map[string]interface{}); ok {
-				result.Port = convertPortFromValue(port["port"].(float64))
+				result.Port = convertPortFromValue(port["port"])
 				if result.Port == 0 {
 					continue
 				}

